stdlib/public/runtime/Reflection.mm

```diff
@@ -1068,62 +1068,57 @@ void swift_ObjCMirror_subscript(String *outString,
   
 // -- MagicMirror implementation.
 
-// TODO: There are other non-Apple platforms that underscore asm symbols.
-#if defined(__APPLE__)
-# define UNDERSCORE "_"
-#else
-# define UNDERSCORE
-#endif
+#define GLUE_EXPANDED(a,b) a##b
+#define GLUE(a,b) GLUE_EXPANDED(a,b)
+#define SYMBOL_NAME(name) GLUE(__USER_LABEL_PREFIX__,name)
```

Comment:
I'm not sure that using a non-portable user label prefix is a net portability win. @tinysun212 is actively working on MSVC support. I think we should keep the conditional, and use `__USER_LABEL_PREFIX__` if it's available:

```
#if defined(__USER_LABEL_PREFIX__)
# define USER_LABEL_PREFIX __USER_LABEL_PREFIX
#else
# error "unimplemented"
#endif
```